buildspec-deploy.yml

```diff
@@ -0,0 +1,29 @@
+version: 0.2
+
+phases:
+    install:
+      runtime-versions:
+        nodejs: 14
+        python: 3.8
+      commands:
+        - env
+        - npm install -g yarn
+        - pip3 install --upgrade pip
+        - pip3 install --upgrade awscli
+        - apt install jq -y
+    build:
+      commands:
+        - ls -al
+        - tar -xzf fhirworks.tgz
+        - ( cd $CODEBUILD_SRC_DIR_DepCog && cat cognito-deploy-output.json )
+        - ( cd $CODEBUILD_SRC_DIR_DepCogIDP && cat cognito-idp-deploy-output.json )
+        - POOL=$(cd $CODEBUILD_SRC_DIR_DepCog && jq -r '.CloudDataHubUserPoolId' cognito-deploy-output.json) && echo "POOL= $POOL"
+        - POOL_CLIENT=$(cd $CODEBUILD_SRC_DIR_DepCogIDP && jq -r '.CloudDataHubUserPoolClientId' cognito-idp-deploy-output.json) && echo "POOL_CLIENT= $POOL_CLIENT"
+        - echo "POOL_DOMAIN= $POOL_DOMAIN"
+        - (cd fhirworks/fhir-works-on-aws-deployment/scripts && ./install.sh --silentInstall "yes" --stage "$STAGE_NAME" --stagetype "$STAGE_TYPE" --region "$REGION" --pool "$POOL" --poolclient "$POOL_CLIENT" --pooldomain "$POOL_DOMAIN" --multiTenancyStrategy "UrlBased" --multiTenancyTokenClaim "cognito:groups" --corsOrigins "ALL_ORIGINS" --useApiKeys "no" --multiTenancyTokenClaimValuePrefix "quentry:")
+        - |
+          echo "{ \"ServiceEndpoint\": \"$(grep 'ServiceEndpoint: ' fhirworks/fhir-works-on-aws-deployment/Info_Output.yml | sed 's/^.*: //')\" }" > fw-deploy-output.json
```

Comment:
Use appropriate tool for processing YAML, not `grep` and `sed`. Don't learn from AWS "Solutions".

Reply:
Unfortunately I can because the content of that file (although having extension  .yml) is not a valid yml file... thank you aws guys...

```
yarn run v1.22.10
$ /codebuild/output/src659/src/s3/00/fhirworks/fhir-works-on-aws-deployment/node_modules/.bin/serverless info --verbose --region us-west-2 --stage ricardo-fhirworks --stageType dev --multiTenancyStrategy UrlBased --multiTenancyTokenClaim cognito:groups --multiTenancyTokenClaimValuePrefix quentry: --corsOrigins ALL_ORIGINS --useApiKeys false
Serverless: Configuration warning at 'provider.environment.VALIDATOR_LAMBDA_ALIAS': unsupported configuration format
Serverless:  
Serverless: Learn more about configuration validation here: http://slss.io/configuration-validation
Serverless:  
Service Information
service: fhir-service
stage: ricardo-fhirworks
region: us-west-2
stack: fhir-service-ricardo-fhirworks
resources: 97
api keys:
  None
endpoints:
  ANY - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks
  ANY - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/{proxy+}
  GET - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/metadata
  OPTIONS - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks
  OPTIONS - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/{proxy+}
  OPTIONS - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/metadata
functions:
  fhirServer: fhir-service-ricardo-fhirworks-fhirServer
  ddbToEs: fhir-service-ricardo-fhirworks-ddbToEs
  startExportJob: fhir-service-ricardo-fhirworks-startExportJob
  stopExportJob: fhir-service-ricardo-fhirworks-stopExportJob
  getJobStatus: fhir-service-ricardo-fhirworks-getJobStatus
  updateStatus: fhir-service-ricardo-fhirworks-updateStatus
  uploadGlueScripts: fhir-service-ricardo-fhirworks-uploadGlueScripts
layers:
  None

Stack Outputs
ResourceDynamoDBTableV2StreamArn: arn:aws:dynamodb:us-west-2:657017214954:table/resource-db-ricardo-fhirworks/stream/2021-07-09T21:53:10.067
ElasticSearchKibanaUserPoolId: us-west-2_e21qAzX6t
ElasticSearchKibanaUserPoolAppClientId: 7p4gnrtr8mqvuoo1mo2gs6ctsk
StartExportJobLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-startExportJob:26
UploadGlueScriptsLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-uploadGlueScripts:27
BulkExportStateMachineArn: arn:aws:states:us-west-2:657017214954:stateMachine:BulkExportStateMachine-gbrBgmkvOHWm
FHIRBinaryBucket: fhir-service-ricardo-fhirworks-fhirbinarybucket-1b0vrh4uqx6a2
ElasticSearchDomainKibanaEndpoint: https://search-fhir-se-elasti-1i7bd74i4mnbe-fzxo5m36igibausj6hidz2ayxu.us-west-2.es.amazonaws.com/_plugin/kibana
ElasticSearchDomainEndpoint: https://search-fhir-se-elasti-1i7bd74i4mnbe-fzxo5m36igibausj6hidz2ayxu.us-west-2.es.amazonaws.com
ServerlessDeploymentBucketName: fhir-service-ricardo-fhi-serverlessdeploymentbuck-t4xqscyx1qw3
ExportRequestDynamoDBTableArn: arn:aws:dynamodb:us-west-2:657017214954:table/export-request-ricardo-fhirworks
CloudwatchExecutionLogGroupArn: arn:aws:logs:us-west-2:657017214954:log-group:API-Gateway-Execution-Logs_457ufz0tca/ricardo-fhirworks:log-stream:*
UpdateStatusLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-updateStatus:27
FhirServerLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-fhirServer:24
DdbToEsLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-ddbToEs:26
UserPoolId: us-west-2_s57V70qEK
UserPoolAppClientId: 7eojkkcnpu8o2ug0e3iq0obkqr
ResourceDynamoDBTableV2Arn: arn:aws:dynamodb:us-west-2:657017214954:table/resource-db-ricardo-fhirworks
GetJobStatusLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-getJobStatus:26
StopExportJobLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-stopExportJob:26
CloudwatchExecutionLogGroup: API-Gateway-Execution-Logs_457ufz0tca/ricardo-fhirworks
ServiceEndpoint: https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks

Done in 2.73s.yarn run v1.22.10
$ /codebuild/output/src659/src/s3/00/fhirworks/fhir-works-on-aws-deployment/node_modules/.bin/serverless info --verbose --region us-west-2 --stage ricardo-fhirworks --stageType dev --multiTenancyStrategy UrlBased --multiTenancyTokenClaim cognito:groups --multiTenancyTokenClaimValuePrefix quentry: --corsOrigins ALL_ORIGINS --useApiKeys false
Serverless: Configuration warning at 'provider.environment.VALIDATOR_LAMBDA_ALIAS': unsupported configuration format
Serverless:  
Serverless: Learn more about configuration validation here: http://slss.io/configuration-validation
Serverless:  
Service Information
service: fhir-service
stage: ricardo-fhirworks
region: us-west-2
stack: fhir-service-ricardo-fhirworks
resources: 97
api keys:
  None
endpoints:
  ANY - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks
  ANY - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/{proxy+}
  GET - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/metadata
  OPTIONS - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks
  OPTIONS - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/{proxy+}
  OPTIONS - https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks/metadata
functions:
  fhirServer: fhir-service-ricardo-fhirworks-fhirServer
  ddbToEs: fhir-service-ricardo-fhirworks-ddbToEs
  startExportJob: fhir-service-ricardo-fhirworks-startExportJob
  stopExportJob: fhir-service-ricardo-fhirworks-stopExportJob
  getJobStatus: fhir-service-ricardo-fhirworks-getJobStatus
  updateStatus: fhir-service-ricardo-fhirworks-updateStatus
  uploadGlueScripts: fhir-service-ricardo-fhirworks-uploadGlueScripts
layers:
  None

Stack Outputs
ResourceDynamoDBTableV2StreamArn: arn:aws:dynamodb:us-west-2:657017214954:table/resource-db-ricardo-fhirworks/stream/2021-07-09T21:53:10.067
ElasticSearchKibanaUserPoolId: us-west-2_e21qAzX6t
ElasticSearchKibanaUserPoolAppClientId: 7p4gnrtr8mqvuoo1mo2gs6ctsk
StartExportJobLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-startExportJob:26
UploadGlueScriptsLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-uploadGlueScripts:27
BulkExportStateMachineArn: arn:aws:states:us-west-2:657017214954:stateMachine:BulkExportStateMachine-gbrBgmkvOHWm
FHIRBinaryBucket: fhir-service-ricardo-fhirworks-fhirbinarybucket-1b0vrh4uqx6a2
ElasticSearchDomainKibanaEndpoint: https://search-fhir-se-elasti-1i7bd74i4mnbe-fzxo5m36igibausj6hidz2ayxu.us-west-2.es.amazonaws.com/_plugin/kibana
ElasticSearchDomainEndpoint: https://search-fhir-se-elasti-1i7bd74i4mnbe-fzxo5m36igibausj6hidz2ayxu.us-west-2.es.amazonaws.com
ServerlessDeploymentBucketName: fhir-service-ricardo-fhi-serverlessdeploymentbuck-t4xqscyx1qw3
ExportRequestDynamoDBTableArn: arn:aws:dynamodb:us-west-2:657017214954:table/export-request-ricardo-fhirworks
CloudwatchExecutionLogGroupArn: arn:aws:logs:us-west-2:657017214954:log-group:API-Gateway-Execution-Logs_457ufz0tca/ricardo-fhirworks:log-stream:*
UpdateStatusLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-updateStatus:27
FhirServerLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-fhirServer:24
DdbToEsLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-ddbToEs:26
UserPoolId: us-west-2_s57V70qEK
UserPoolAppClientId: 7eojkkcnpu8o2ug0e3iq0obkqr
ResourceDynamoDBTableV2Arn: arn:aws:dynamodb:us-west-2:657017214954:table/resource-db-ricardo-fhirworks
GetJobStatusLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-getJobStatus:26
StopExportJobLambdaFunctionQualifiedArn: arn:aws:lambda:us-west-2:657017214954:function:fhir-service-ricardo-fhirworks-stopExportJob:26
CloudwatchExecutionLogGroup: API-Gateway-Execution-Logs_457ufz0tca/ricardo-fhirworks
ServiceEndpoint: https://457ufz0tca.execute-api.us-west-2.amazonaws.com/ricardo-fhirworks

Done in 2.73s.
```

Reply:
@organom , please prepare and have `Info_Output.yml` handy. I want to look at it. It's probably a bug, which upstream should fix.